tooling/log-redaction/main.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+
+	_ "embed"
+
+	"github.com/Azure/ARO-Tools/pkg/config"
+	"github.com/Azure/ARO-Tools/pkg/config/ev2config"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+//go:embed config.tmpl
+var mustGatherTemplate []byte
+
+type redactionConfig map[string][]any
+
+type Options struct {
+	ServiceConfigFile   string
+	RedactionConfigFile string
+	Cloud               string
+	Environment         string
+	Region              string
+	Stamp               int
+	Ev2Cloud            string
+	RegionShortSuffix   string
+	Debug               bool
+}
+
+func main() {
+	if os.Getenv("DEBUG") == "true" {
+		logrus.SetLevel(logrus.DebugLevel)
+	}
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
+	defer stop()
+
+	cmd := &cobra.Command{
+		Use:              "log-redaction",
+		Short:            "Extract sensitive values from config for log redaction",
+		Long:             "Reads redaction keys from config.redaction.yaml and extracts their actual values from the main config file",
+		SilenceUsage:     true,
+		TraverseChildren: true,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			cmd.SetContext(ctx)
+		},
+	}
+
+	opts := &Options{
+		Stamp: 1,
+	}
+
+	cmd.Flags().StringVar(&opts.ServiceConfigFile, "service-config-file", "config/config.yaml", "Path to the service configuration file.")
+	cmd.Flags().StringVar(&opts.RedactionConfigFile, "redaction-config-file", "config/config.redaction.yaml", "Path to the redaction configuration file.")
+	cmd.Flags().StringVar(&opts.Cloud, "cloud", "dev", "The name of the cloud.")
+	cmd.Flags().StringVar(&opts.Environment, "environment", "dev", "The name of the environment.")
+	cmd.Flags().StringVar(&opts.Region, "region", "eastus2", "The name of the region.")
+	cmd.Flags().IntVar(&opts.Stamp, "stamp", opts.Stamp, "Stamp value to use.")
```

Comment:
what is a stamp?

Reply:
When some deployment unit can be replicated in a region many times, the stamp number is used to differentiate between the replicas. I don't imagine this will need to be set by the user of this tool, we could resolve the config for stamp 0 without loss of generality

Reply:
I'll remove this parameter